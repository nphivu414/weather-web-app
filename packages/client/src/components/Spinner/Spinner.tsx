import * as React from 'react';

export type SpinnerProps = {
  isSpinning?: boolean;
};

const Spinner: React.FC<SpinnerProps> = ({ isSpinning = true }) => {
  return (
    <>
      {isSpinning && (
        <div className="d-flex justify-content-center">
          <div className="spinner-border text-primary" role="status">
            <span className="visually-hidden">Loading...</span>
          </div>
        </div>
      )}
    </>
  );
};

export default Spinner;
