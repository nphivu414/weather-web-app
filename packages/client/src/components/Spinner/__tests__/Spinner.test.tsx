import * as React from 'react';
import { shallow } from 'enzyme';

import Spinner, { SpinnerProps } from '../Spinner';

const props: SpinnerProps = {
  isSpinning: false,
};

describe('Spinner', () => {
  it('should match a snapshot', () => {
    expect(shallow(<Spinner {...props} />)).toMatchSnapshot();
  });
  it('should show spinner', () => {
    const wrapper = shallow(<Spinner {...props} isSpinning />);
    expect(wrapper.find('div.spinner-border')).toHaveLength(1);
  });
});
