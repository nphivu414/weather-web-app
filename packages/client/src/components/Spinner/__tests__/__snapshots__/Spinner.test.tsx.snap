// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should match a snapshot 1`] = `ShallowWrapper {}`;
